### DISCUSSION TITLE
Balance number formatting

### DISCUSSION THREAD

**Discussion by yudhistiraindyka** [thejoabo/virtualfisher-bot#17] [Ideas]
Hello @thejoabo, I was playing around earlier and noticed the number formatting for balance. I realized that after reaching certain levels, it's not unusual to have a couple Billions. I don't know if it's just me but the exponential number format shown here is not as easy to tell at a glance how much money we currently have:
![image](https://user-images.githubusercontent.com/33085123/181022129-0e05c79a-397a-41b1-9d6f-7c4d0efe3ffc.png)

Therefore I propose 2 solutions, the first one is just showing the whole value as is and use comma as a separator, while the second one would convert the value into strings, take the first 3 digits and append "Billion" at the end. Examples:
![image](https://user-images.githubusercontent.com/33085123/181022613-5871ea01-d73d-4d96-827e-1165e80893ed.png)
![image](https://user-images.githubusercontent.com/33085123/181024014-f657b689-d15c-4c33-b0fb-1a89b490d5fa.png)

What do you think about this? For me personally, I prefer the first solution since it's easier to read. I'd post a pull request later once you approved.

**Comment by thejoabo** [comment]
Hello @yudhistiraindyka,

I thought in scientific notation because it was universal and shortens the number (so even in larger numbers would fit the screen, since it follows a limited length). 

But I agree, it can sometimes be awful to figure out what it means at a quick glance. I **think that we could add an option in the config file for that, and call it "number_formatting" and set the default value to display your first solution**. So the user can decide which one he likes the most (and would affects all numbers on the screen - like the xp value on the %f command and the xp to the next level). 

The only place where I can't remove completely the notation approach is in the compact mode that I'm working on. But on the standard mode shouldn't be a problem.

Anyway, should I add this new parameter in the config file or just change the formatting it in the source code ? What do you think its better ?

**Comment by yudhistiraindyka** [comment]
Hello @thejoabo, I believe that adding an option in the config file would be the way to go, I dunno why I didn't think of that before.

**Reply by thejoabo** [reply]
Alright then, will be added on the next update ! Thanks for your suggestion.
